File: SDKs/figshare.com/2.0.0/python/src/sdk/models/operations/article_details.py
import dataclasses
from typing import Optional
from ..shared import articlecomplete as shared_articlecomplete


@dataclasses.dataclass
class ArticleDetailsPathParams:
    article_id: int = dataclasses.field(metadata={'path_param': { 'field_name': 'article_id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class ArticleDetailsRequest:
    path_params: ArticleDetailsPathParams = dataclasses.field()
    

@dataclasses.dataclass
class ArticleDetailsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    article_complete: Optional[shared_articlecomplete.ArticleComplete] = dataclasses.field(default=None)
    

File: SDKs/figshare.com/2.0.0/typescript/src/sdk/models/operations/articledetails.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ArticleDetailsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=article_id" })
  articleId: number;
}


export class ArticleDetailsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ArticleDetailsPathParams;
}


export class ArticleDetailsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  articleComplete?: shared.ArticleComplete;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
